core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1322,4 +1336,38 @@ object Chunk {
     private val CharClass       = classTag[Char]
     private val CharClassBox    = classTag[java.lang.Character]
   }
+
+  final case class AnyRefArray[A <: AnyRef](array: Array[A]) extends Arr[A]
+
+  final case class ByteArray(array: Array[Byte]) extends Arr[Byte] {
+    override def byte(index: Int)(implicit ev: Byte <:< Byte): Byte = array(index)
+  }
+
+  final case class CharArray(array: Array[Char]) extends Arr[Char] {
+    override def char(index: Int)(implicit ev: Char <:< Char): Char = array(index)
+  }
+
+  final case class IntArray(array: Array[Int]) extends Arr[Int] {
+    override def int(index: Int)(implicit ev: Int <:< Int): Int = array(index)
+  }
+
+  final case class LongArray(array: Array[Long]) extends Arr[Long] {
+    override def long(index: Int)(implicit ev: Long <:< Long): Long = array(index)
+  }
+
+  final case class DoubleArray(array: Array[Double]) extends Arr[Double] {
+    override def double(index: Int)(implicit ev: Double <:< Double): Double = array(index)
+  }
+
+  final case class FloatArray(array: Array[Float]) extends Arr[Float] {
+    override def float(index: Int)(implicit ev: Float <:< Float): Float = array(index)
+  }
+
+  final case class ShortArray(array: Array[Short]) extends Arr[Short] {
+    override def short(index: Int)(implicit ev: Short <:< Short): Short = array(index)
```

Comment:
Really nice to see all these box-free accessors!! ❤️

Reply:
Is there a note anywhere on this technique for avoiding boxing? I don't understand what role the implicit is serving here.

Reply:
The implicit acts as proof in the parent type that it is safe to call this method in the child type. It's a way of exposing the specialized method in the parent.